subsys/bluetooth/host/crypto.c

```diff
@@ -114,6 +115,40 @@ int bt_rand(void *buf, size_t len)
 
 	return -EIO;
 }
+#else /* !CONFIG_BT_HOST_CRYPTO_RANDOM */
+int bt_rand(void *buf, size_t len)
+{
+	int ret, size;
+	size_t i = 0;
+
+	/* Check first that HCI_LE_Rand is supported */
+	if (!BT_CMD_TEST(bt_dev.supported_commands, 27, 7)) {
+		return -ENOTSUP;
+	}
+
+	while (len) {
+		struct bt_hci_rp_le_rand *rp;
+		struct net_buf *rsp;
+
+		ret = bt_hci_cmd_send_sync(BT_HCI_OP_LE_RAND, NULL, &rsp);
+		if (ret) {
+			return ret;
+		}
+
+		size = MIN(len, 8);
```

Comment:
Please avoid magic numbers

```suggestion
		size = MIN(len, sizeof(rsp->rand));
```